File: tests/java/src/main/java/com/escape/tests/cases/Case006CacheSubscription.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 006: cache subscription payload for downstream business processing. */
public class Case006CacheSubscription {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: cache subscription records and prepare transport-ready payload.
        String taskName = "cache_subscription";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "subscription");
        payload.put("stage", "cache");
        payload.put("input", raw);
        // ESCAPE: payload is promoted to class-level retained cache.
        RETAINED_CACHE.put("case_006", payload);
        return "ok";
    }
}

File: tests/go/cases/case_006_cache_subscription.go
package escape_tests

var retainedCase006 = []map[string]string{}

func Case006CacheSubscription(input string) string {
    // Task case 006: cache subscription payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "cache_subscription",
        "entity": "subscription",
        "stage": "cache",
        "input": raw,
    }
    // ESCAPE: payload is promoted to package-level retained cache.
    retainedCase006 = append(retainedCase006, payload)
    return "ok"
}
